node/src/Pos/Delegation/Cede/Holders.hs

```diff
@@ -59,12 +60,18 @@ evalMapCede = flip Ether.evalLazyStateT
 
 instance MonadDBRead m => MonadCedeRead (MapCede m) where
     getPsk iPk =
-        ether $ use (at iPk) >>= \case
+        ether $ use (cmPskMods . at iPk) >>= \case
             Nothing                -> lift $ DB.getPskByIssuer $ Right iPk
             Just (DlgEdgeDel _)    -> pure Nothing
             Just (DlgEdgeAdd psk ) -> pure (Just psk)
+    hasPostedThisEpoch sId =
+        ether $ use (cmHasPostedThisEpoch . at sId) >>= \case
+            Nothing                -> lift $ DB.isIssuerPostedThisEpoch sId
+            Just v                 -> pure v
 
 instance MonadDBRead m => MonadCede (MapCede m) where
     modPsk eAction = do
         let issuer = dlgEdgeActionIssuer eAction
-        ether $ identity %= HM.insert issuer eAction
+        ether $ cmPskMods %= HM.insert issuer eAction
+    addThisEpochPosted sId = ether $ cmHasPostedThisEpoch %= HM.insert sId True
```

Comment:
why didn't you use `Set` instead of `Map` of Bool?

Reply:
ok I got it, IMHO it would be better to use `MapModifier Stakeholer ()` b/c such construction is used for similar things and hence it's more intuitive, but it's subjective.